### ISSUE TITLE
[NextJS] Use web frameworks' built-in auth cookie instead of custom implementation

### ISSUE THREAD

**Issue by jhuleatt** [firebase/friendlyeats-web#243]
The Firebase CLI's web frameworks tooling sets an auth cookie: https://github.com/FirebaseExtended/firebase-framework-tools/blob/087035d86c65c5810ab71f889e4d6160398368aa/src/firebase-aware.ts#L32

This means we probably don't need the current custom implementation: https://github.com/firebase/friendlyeats-web/blob/c43de974fbf44151217fe3238d610edee7b86b28/nextjs-end/src/app/api/route.js#L9

**Comment by ismaelaarab** [comment]
@jhuleatt 

Does this imply that the example is primarily designed for use with Firebase Hosting?

If someone chooses to host their project on Vercel instead, will they need to implement additional custom logic in order to set the session on the server side?

**Comment by matallui** [comment]
Same question? ☝🏼

**Comment by leonam-okajima** [comment]
We really need clarification on this. Are the example and tutorial designed only for use with Firebase Hosting?

**Comment by EmanuelU** [comment]
after a lot of digging I am fairly sure the __session cookie expected in the tutorial is a firebase only thing, not from the auth package but from the frameworks.

**Comment by josancamon19** [comment]
Yup, thanks for asking this, I was close to giving up already, and just verified that indeed, it only works if you use firebase, either from emulators or not.

**Comment by Herohtar** [comment]
This issue should be marked as solved, because the file in question no longer exists and indeed it seems that they implemented the auth token parsing using the `__session` cookie:
https://github.com/firebase/friendlyeats-web/blob/02be588cf1ca3db5e8d7296b6aae19681f2533cf/nextjs-end/src/lib/firebase/firebase.js#L27

However, as a couple of the more recent comment indicate, the `__session` cookie doesn't actually appear to be set anywhere in my own project, even when using Firebase Hosting. I'm not sure if this is something that changed with Firebase Auth or some other issue.

I did notice that something that looks like an auth token appears to be stored in Indexed DB, so maybe that's where it is now. I'm not sure how you make use of that though.

**Comment by raky291** [comment]
The Firebase team should improve the documentation and the examples. The documentation is scattered across multiple pages and isn't very clear.

We would greatly appreciate it if you could vote for better documentation here: https://firebase.uservoice.com/forums/948424-general/suggestions/48594224-full-example-of-firebaseserverapp-in-next-js-14-us
